src/main/webapp/WEB-INF/tags/shared/commentsPanel.tag

```diff
@@ -1,27 +1,198 @@
 <%@ tag description="Panel for student comments" %>
 <%@ taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c" %>
+<%@ taglib uri="http://java.sun.com/jsp/jstl/functions" prefix="fn" %>
 <%@ taglib tagdir="/WEB-INF/tags/shared" prefix="shared" %>
+<%@ tag import="teammates.common.util.Const" %>
+<%@ tag import="teammates.common.datatransfer.CommentParticipantType" %>
 <%@ attribute name="commentsForStudentsTables" type="java.util.Collection" required="true" %>
 <%@ attribute name="courseId" %>
 <%@ attribute name="viewingDraft" %>
+<%@ attribute name="forRecordsPage" %>
+<%@ attribute name="newCommentIndex" %>
+<c:choose>
+    <c:when test="${viewingDraft}">
+        <c:set var="bodyTitle" value="Comment drafts" />
+    </c:when>
+    <c:when test="${forRecordsPage}">
+        <c:set var="bodyTitle" value="Comments for ${data.studentName}" />
+    </c:when>
+    <c:otherwise>
+        <c:set var="bodyTitle" value="Comments for students" />
+    </c:otherwise>
+</c:choose>
 <div class="panel panel-primary">
     <div class="panel-heading">
-        <strong>${viewingDraft ? 'Comment drafts' : 'Comments for students'}</strong>
+        <strong>${bodyTitle}</strong>
     </div>
     <div class="panel-body">
         <c:if test="${viewingDraft}">Your comments that are not finished:</c:if>
         <c:set var="commentIndex" value="${0}"/>
         <c:forEach items="${commentsForStudentsTables}" var="commentsForStudentsTable">
             <div class="panel panel-info student-record-comments${commentsForStudentsTable.extraClass}"
-                 <c:if test="${empty commentsForStudentsTable.rows}">style="display: none;"</c:if>>
+                 <c:if test="${empty commentsForStudentsTable.rows && not forRecordsPage}">style="display: none;"</c:if>>
```

Comment:
While it may change the HTML output, should we use an if to see if we should even render it at all?
May be better than a separate issue, though, I guess.